module_bom_test.go

```diff
@@ -143,13 +149,200 @@ func testModuleBOM(t *testing.T, context spec.G, it spec.S) {
 						PURL:     "pkg:npm/rightpad@1.0.0",
 						Licenses: []string{"Apache"},
 						Version:  "1.0.0",
+						Checksum: packit.BOMChecksum{
+							Algorithm: packit.SHA256,
+							Hash:      "123456789",
+						},
 					},
 				},
 			}))
 
 			Expect(filepath.Join(workingDir, "bom.json")).ToNot(BeAnExistingFile())
 		})
 
+		context("the bom.json does not contain hashes for node_modules", func() {
+			it.Before(func() {
+				executable.ExecuteCall.Stub = func(execution pexec.Execution) error {
+					Expect(os.WriteFile(filepath.Join(workingDir, "bom.json"), []byte(
+						`
+{
+  "components": [
+    {
+      "type": "library",
+      "name": "leftpad",
+      "version": "0.0.1",
+      "description": "left pad numbers",
+      "licenses": [
+        {
+          "license": {
+            "id": "BSD-3-Clause"
+          }
+        }
+      ],
+      "purl": "pkg:npm/leftpad@0.0.1"
+		}
+  ]
+}
+`), 0600)).To(Succeed())
```

Comment:
FYI, JSON is whitespace agnostic and so this raw string doesn't need to be outdented all the way to the line start. Its perfectly reasonable to align it with the rest of your code. It will probably help with the legibility of this set of tests. You can see this in quite a few other codebases: https://github.com/paketo-buildpacks/npm-start/blob/cd8145f9038dc0eec655ecd3e68cd43a39eaed8f/build_test.go#L44-L50.